examples/js/loaders/OBJLoader2.js

```diff
@@ -0,0 +1,1002 @@
+					break;
+
+				case Consts.LINE_O:
+					if ( this.rawObject.vertices.length > 0 ) {
+
+						this._processCompletedObject( buffer[ 1 ], null );
+						reachedFaces = false;
+
+					} else {
+
+						this.rawObject._pushObject( buffer[ 1 ] );
+
+					}
+					break;
+
+				case Consts.LINE_MTLLIB:
+					this.rawObject._pushMtllib( buffer[ 1 ] );
+					break;
+
+				case Consts.LINE_USEMTL:
+					this.rawObject._pushUsemtl( buffer[ 1 ] );
+					break;
+
+				default:
+					break;
+			}
+			return reachedFaces;
+		};
+
+		Parser.prototype._processCompletedObject = function ( objectName, groupName ) {
+			this.rawObject._finalize( this.meshCreator, this.inputObjectCount, this.debug );
+			this.inputObjectCount++;
+			this.rawObject = this.rawObject._newInstanceFromObject( objectName, groupName );
+		};
+
+		Parser.prototype._processCompletedGroup = function ( groupName ) {
+			var notEmpty = this.rawObject._finalize( this.meshCreator, this.inputObjectCount, this.debug );
+			if ( notEmpty ) {
+
+				this.inputObjectCount ++;
+				this.rawObject = this.rawObject._newInstanceFromGroup( groupName );
+
+			} else {
+
+				// if a group was set that did not lead to object creation in finalize, then the group name has to be updated
+				this.rawObject._pushGroup( groupName );
+
+			}
+		};
+
+		Parser.prototype._finalize = function () {
+			this.rawObject._finalize( this.meshCreator, this.inputObjectCount, this.debug );
+			this.inputObjectCount++;
+		};
+
+		return Parser;
+	})();
+
+	/**
+	 * {@link RawObject} is only used by {@link Parser}.
+	 * The user of OBJLoader2 does not need to care about this class.
+	 * It is defined publicly for inclusion in web worker based OBJ loader ({@link THREE.OBJLoader2.WWOBJLoader2})
+	 */
+	var RawObject = (function () {
+
+		function RawObject( objectName, groupName, mtllibName ) {
+			this.globalVertexOffset = 1;
+			this.globalUvOffset = 1;
+			this.globalNormalOffset = 1;
+
+			this.vertices = [];
+			this.normals = [];
+			this.uvs = [];
+
+			// faces are stored according combined index of group, material and smoothingGroup (0 or not)
+			this.mtllibName = ( mtllibName != null ) ? mtllibName : 'none';
+			this.objectName = ( objectName != null ) ? objectName : 'none';
+			this.groupName = ( groupName != null ) ? groupName : 'none';
+			this.activeMtlName = 'none';
+			this.activeSmoothingGroup = 1;
+
+			this.mtlCount = 0;
+			this.smoothingGroupCount = 0;
+
+			this.rawObjectDescriptions = [];
+			// this default index is required as it is possible to define faces without 'g' or 'usemtl'
+			var index = this._buildIndex( this.activeMtlName, this.activeSmoothingGroup );
+			this.rawObjectDescriptionInUse = new RawObjectDescription( this.objectName, this.groupName, this.activeMtlName, this.activeSmoothingGroup );
+			this.rawObjectDescriptions[ index ] = this.rawObjectDescriptionInUse;
+		}
+
+		RawObject.prototype._buildIndex = function ( materialName, smoothingGroup) {
```

Comment:
`RawObject` and `MeshCreator` were initially public and all non-public methods were underscored. This is no longer necessary and it is likely misleading. I was just lazy. Should be fairly easy to adjust... Give me half an hour.
Thanks for changing the title, btw. 😉